src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -397,18 +417,32 @@ func (d *downsamplerAndWriter) writeAggregatedBatch(
 
 		samplesAppender, err := appender.SamplesAppender(opts)
 		if err != nil {
-			return err
+			multiErr = multiErr.Add(err)
+		}
+
+		if appender.IsDropPolicyApplied() {
+			iter.SetCurrentMetadata(ts.Metadata{true})
 		}
 
 		for _, dp := range datapoints {
 			err := samplesAppender.AppendGaugeTimedSample(dp.Timestamp, dp.Value)
 			if err != nil {
-				return err
+				// If we see an error break out so we can try processing the
+				// next datapoint.
+				multiErr = multiErr.Add(err)
+				break
 			}
 		}
 	}
 
-	return iter.Error()
+	if err := iter.Error(); err != nil {
+		multiErr.Add(err)
+	}
+
+	if multiErr.NumErrors() == 0 {
+		return nil
+	}
+	return multiErr
```

Comment:
Instead of checking NumErrors can just return `multiErr.FinalError()` which will return nil or the concat'd errors.